include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,199 @@
+#include <eve/detail/kumi.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/module/core.hpp>
+#include <eve/product_type.hpp>
+#include <cstddef>
+#include <type_traits>
+#include <memory>
+
+namespace eve::algo::detail
+{
+  struct memory_helper
+  {
+    static void copy_all_aligned(std::byte const* src, std::byte* dst, std::size_t size)
+    {
+      auto aligned_src = eve::as_aligned(reinterpret_cast<std::uint8_t const*>(src), lane<64>);
+      auto aligned_dst = eve::as_aligned(reinterpret_cast<std::uint8_t*>(dst), lane<64>);
+      eve::algo::copy(as_range(aligned_src, aligned_src+size), aligned_dst);
+    }
+
+    template<typename T> static EVE_FORCEINLINE auto as_aligned_pointer(T* ptr)
+    {
+      return eve::as_aligned(ptr, eve::detail::cache_line_cardinal<T>{});
+    }
+
+    template<typename T>
+    static  auto realign(std::size_t n) { return eve::align(sizeof(T)*n, eve::over{64}); }
+
+    template<typename Storage>
+    static auto allocate(Storage const& s, std::size_t n)
+    {
+      using storage_t = typename Storage::storage_t;
+      auto a = s.get_allocator();
+      return storage_t{ (typename Storage::byte_t*)(a.allocate_aligned(n,64)), a};
+     }
+  };
+
+  template<eve::product_type Type, eve::simd_allocator Allocator>
+  struct soa_storage
+  {
+    using is_product_type = void;
+    using value_type = Type;
+
+    soa_storage() noexcept
+            : indexes_{}, storage_( nullptr, aligned_deleter{Allocator{}} ), capacity_{}
+    {}
+
+    explicit  soa_storage(Allocator a) noexcept
+            : indexes_{}, storage_( nullptr, aligned_deleter{a} ), capacity_{}
+    {}
+
+    soa_storage(Allocator a, std::size_t c) : soa_storage(a)
+    {
+      // Strong guarantee on allocation
+      auto local = memory_helper::allocate(*this, byte_size(c));
+      storage_.swap(local);
+
+      capacity_ = aligned_capacity(c);
+
+      auto        sub     = storage_.get();
+      std::size_t offset  = 0ULL;
+
+      kumi::for_each_index( [&]<typename Idx>(Idx, auto& s)
+                            {
+                              // Start current sub-span data lifetime
+                              using type  = kumi::element_t<Idx::value,Type>;
+                              auto sz     = memory_helper::realign<type>(c);
+                              std::memmove(reinterpret_cast<type*>(sub+offset), sub+offset, sz);
+
+                              // Update the index
+                              s       = offset;
+                              offset += sz;
+                            }
+                          , indexes_
+                          );
+    }
+
+    soa_storage(soa_storage const& src) : soa_storage(src.storage_.get_deleter())
+    {
+      auto sz = byte_size(src.capacity_);
+
+      // Strong guarantee on allocation
+      auto local  = memory_helper::allocate(*this,sz);
+      memory_helper::copy_all_aligned(src.storage_.get(), local.get(), sz );
+      storage_.swap(local);
```

Comment:
move

Reply:
move and swap do the exact same thing so I am a bit confused about your insistence on this change.